base/common/src/main/java/com/netscape/certsrv/profile/ProfilePolicy.java

```diff
@@ -108,19 +120,239 @@ public boolean equals(Object obj) {
+                descriptionElement.appendChild(document.createTextNode(constraint.getText()));
+                pcElement.appendChild(descriptionElement);
+            }
+            if (constraint.getClassId() != null) {
+                Element classIdElement = document.createElement("classId");
+                classIdElement.appendChild(document.createTextNode(constraint.getClassId()));
+                pcElement.appendChild(classIdElement);
+            }
+            if (constraint.getConstraints() != null) {
+                for (PolicyConstraintValue pcv : constraint.getConstraints()) {
+                    Element constraintElement = document.createElement("constraint");
+                    constraintElement.setAttribute("id", pcv.getName());
+                    Descriptor descriptor = pcv.getDescriptor();
+                    if (descriptor != null) {
+                        Element descriptorElement = document.createElement("descriptor");
+                        if (descriptor.getSyntax() != null) {
+                            Element syntaxElement = document.createElement("mSyntax");
+                            syntaxElement.appendChild(document.createTextNode(descriptor.getSyntax()));
+                            descriptorElement.appendChild(syntaxElement);
+                        }
+                        if (descriptor.getConstraint() != null) {
+                            Element mConstraintElement = document.createElement("mConstraint");
+                            mConstraintElement.appendChild(document.createTextNode(descriptor.getConstraint()));
+                            descriptorElement.appendChild(mConstraintElement);
+                        }
+                        if (descriptor.getDescription(Locale.getDefault()) != null) {
+                            Element descriptionElement = document.createElement("mDescription");
+                            descriptionElement.appendChild(document.createTextNode(descriptor.getDescription(Locale.getDefault())));
+                            descriptorElement.appendChild(descriptionElement);
+                        }
+                        if (descriptor.getDefaultValue() != null) {
+                            Element defaultValueElement = document.createElement("mDef");
+                            defaultValueElement.appendChild(document.createTextNode(descriptor.getDefaultValue()));
+                            descriptorElement.appendChild(defaultValueElement);
+                        }
+                        constraintElement.appendChild(descriptorElement);
+                        if (pcv.getValue() != null) {
+                            Element valueElement = document.createElement("value");
+                            valueElement.appendChild(document.createTextNode(pcv.getValue()));
+                            constraintElement.appendChild(valueElement);
+                        }
```

Comment:
I think this code should be outside of the `if (descriptor != null)`.

Reply:
Good spot, fixed.